### DISCUSSION TITLE
Welcome to DFO R Package Workshop Discussions!

### DISCUSSION THREAD

**Discussion by ateucher** [ateucher/pkg-dev-dfo-2024-11#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to ask questions during the workshop, and discuss ideas with each other. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other workshop participants.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, please introduce yourself:
- Office/department, city/town
- Why you’re here
- Share something you’re good at

**Comment by KeithLewis99** [comment]
Hi Andy - looking forward to the course.  I regularly use Git through RStudio and have a Github account.  I used "usethis::git_sitrep()" and can see my name and email address.  Do I need to do anything further?  Thanks.

**Reply by ateucher** [reply]
Hey @KeithLewis99, thanks for checking in! Great that you have have GitHub all set up. Can you also run `devtools::dev_sitrep()` and see what it says?

**Reply by KeithLewis99** [reply]
Thanks for the quick response Andy.  Here's the output.  Based on below, i'll update devtools.  Please advise if I should do anything else.  Cheers, Keith:
── R ─────────────────────────────────────────────────────────────────
• version: 4.4.1
• path: 'C:/Program Files/R/R-4.4.1/'
• R is out of date (4.4.1 vs 4.4.2)
── Rtools ────────────────────────────────────────────────────────────
• path: 'C:/rtools44/usr/bin/'
── RStudio ───────────────────────────────────────────────────────────
• version: 2024.4.2.764
• RStudio 2024.09.0+375 is now available (you're using 2024.04.2+764).
  Download at: https://www.rstudio.com/products/rstudio/download/
── devtools ──────────────────────────────────────────────────────────
• version: 2.4.5
• devtools or its dependencies out of date:
  'fs', 'glue', 'withr', 'ps', 'xfun'
  Update them with `devtools::update_packages("devtools")`
── dev package ───────────────────────────────────────────────────────
• package: <unset>
• path: <unset>

**Reply by ateucher** [reply]
Thanks Keith. devtools 2.4.5 is the current version, so there are probably minor updates available for those dependencies listed. Not critical to update them, but it's always a good idea!

**Reply by KeithLewis99** [reply]
Thanks!

**Comment by ateucher** [comment]
Andy Teucher, I live in Victoria, BC.

I am here to help people learn to write R packages.

I am good at fixing things

**Comment by boshek** [comment]
Sam Albers. I live in Victoria, BC. Here to get folks excited about R packages. I am good at learning new things.

**Comment by JoleneSutton** [comment]
Hi! I'm really looking forward to this course! I am Jolene, signing in from DFO Gulf, in Moncton NB

**Comment by nickawbrown** [comment]
Hi all, Nick Brown here from Pacific Salmon Stock assessment in BC. I am keen to learn new skills working in R and GitHub. I am good at making plots in R.

**Comment by mathieu-desgagnes** [comment]
Mathieu Desgagnés, Région du Québec

lot of R code here that need packaging!

**Comment by KeithLewis99** [comment]
Keith Lewis, Coastal and Freshwater Ecology Section, St. John's NL.  Do a lot more coding than I was ever trained for - hoping to get better at it.

**Comment by torkjot** [comment]
Hi, I'm Tor. I'm from Nanaimo, BC, stationed at the Pacific Biological Station. I am here to learn about best practices for writing R Packages in order to make them for the Quantitative Assessment Methods Section with DFO. I am good at Badminton.

**Comment by kaylgill** [comment]
Kayleigh Gillespie, I live in Ladysmith, BC. 

I am here to learn more about package development and creating code that is easier to share with others. 

I am good at creating maps :)

**Comment by MMMcMahon** [comment]
Hey All - I'm with DFO Maritimes in the Population Ecology Division  here in Halifax.  I'm from Vancouver originally (also did undergrad in Calgary).  I've made a bunch or R packages, but am never sure if I'm doing them right, hence the course. :)

**Comment by jamiectam** [comment]
Jamie Tam, I am in Halifax, NS.

I'm great at breaking things in R! I am hoping to learn a lot in this course :)

**Comment by adsmithca** [comment]
Hey all, 
Andrew Smith, stock assessment biologist at Institut Maurice Lamontagne in Mont-Joli QC

**Comment by paul-vdb** [comment]
I'm Paul van Dam-Bates. I live in Courtenay BC and work out of the PBS office as a postdoc in Quantitative Assessment Methods.

**Comment by vpourfaraj** [comment]
Vahab Pourfaraj, biologist with DFO. I work out of Nanaimo office in British Columbia. I'm good at making plots.

**Reply by jamiectam** [reply]
Hi Vahab!

**Comment by kateaurora** [comment]
Kate McGivney, I live in Bella Coola, BC.  I am good at adapting other people's code :)

**Comment by CMClark1** [comment]
Hi, I'm Caira Clark, a groundfish biologist at St. Andrews Biological Station in New Brunswick.

**Reply by CMClark1** [reply]
Here to learn how to better put together packages. I'm good at cleaning up and streamlining older scripts.

**Comment by maburton** [comment]
Meghan Burton, I live in Courtenay BC. I am here to learn how to string functions I have floating around into R packages. I'm good at working with Rmarkdown and LaTeX

**Comment by RBelley** [comment]
Rénald Belley, I'm a biologist working on scallop, Arctic Surfclams and macroalgae stock assessments. I'm located in Quebec

**Comment by Nick-Gullage** [comment]
Nick Gullage, I live in St. John's Newfoundland and work with DFO Science.
I'm here to be more efficient and package and re-using my code.
I'm quick to learn new packages/code languages

**Comment by theb8** [comment]
Alexandre Théberge from Qc, looking forward to learn new skills.

**Comment by kirbymorrill** [comment]
Kirby Morrill, I work in Charlo, NB (DFO Gulf). I am here to learn new skills!

**Comment by robynforrest** [comment]
Hello. Robyn Forrest from the Quantitative Assessment Methods Section in Pacific Region (Nanaimo, traditional unceded territory of the Snuneymuxw First Nation). Here to learn how to better organize and share my code. I'm good at growing tomatoes.

**Comment by ManueBSylvestre** [comment]
Manuelle Beaudry-Sylvestre, I live in Sainte-Luce, QC, and I work as an aquatic science biologist at DFO-Mont-Joli. Here to learn an effective workflow for my functions. Good at stealing code from others on github hehe.

**Comment by clairehornby** [comment]
Hi everyone!

My name is Claire, I work as a biologist for DFO based in Winnipeg. Looking forward to learning about package development!

**Comment by ateucher** [comment]
You can write code by surrounding it in "fences" ```

```r
rnorm(56)

1 + 1
```

**Comment by a-potapova** [comment]
Hi I'm Anna, I'm a biologist in the Quantitative Assessment Methods Section and I'm based in Vancouver. Looking forward to improve my project management to make my code more shareable and usable by myself and others!

**Comment by jcosham** [comment]
Hi, I'm Jessica Cosham from Dartmouth Nova Scotia. I'm very good at breaking code! Very excited to get more familiar with GitHub and maybe try my hand at packages in the future.

**Comment by shaniRL** [comment]
Hello! I'm Shani Rousseau from DFO in the Quebec region. Looking forward to learn more about R packaging!

**Comment by sarahgravel** [comment]
Hi all, I'm Sarah Gravel. I'm biologist in the Sclerochronology lab at DFO in Nanaimo, working on some models to improve fish age estimates and species & stock ID from otoliths, and AI for ageing salmon scales. I am currently in Montreal (where I'm from) for a wedding. :) Looking forward to learning about packages!!

**Comment by satchison** [comment]
Hi, Sheila Atchison from Winnipeg here!
I just got back from sea yesterday so bear with me :P

**Comment by munrohannah** [comment]
Hello, Hannah from NL region in St. John's.

**Comment by Fatemehatefi** [comment]
Hello, Fatemeh Hatefi from DFO NL, St. Johns.
